Add X-Plane SDK support

xdarkradx opened the pull request:
Hi,

This commit is for adding the pre-built X-Plane SDK library into VCPKG. I don't have a Mac to test whether I copy the library into the right place or not, so if I'm making any mistake, please let me know. I will try and correct it as soon as possible.

Also, this is my very first pull request for something like this. Please let me know if I'm doing anything wrong.

Thank you so much for your time.

xdarkradx committed:
Add X-Plane SDK support

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5408) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5408) before we can accept your contribution.<br/><br/>:x: xdarkradx [sign now](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5408)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/vcpkg?pullRequest=5408) it.</sub>

Nostracodus committed:
fix: disallow x86 builds

The x-plane SDK website claims that the SDK is only
compatible with x64. The PR that adds support for this
fails in x86, so I'm disallow x86 builds to align with
the x-plane documentation.

Nostracodus committed:
fix: install sdk headers to the root include path

The x-plane SDK website's examples expect that the sdk
headers are at the root of the include path. I'm adjusting
this port to align with those samples, as we want examples
to "just work" as much as possible.

Nostracodus committed:
style: whitespace

Codiferous commented:
Hi, @xdarkradx! Thanks for this PR. Things look pretty good, but there are a few issues we should fix.

Firstly, I found this to fail when building on x86 with errors like so:
```
The following files were built for an incorrect architecture:
    E:/vcpkg-dev2/packages/x-plane_x86-windows/debug/lib/XPLM_64.lib
Expected x86, but was: x64
```

The x-plane SDK website says that the SDK is only usable on 64-bit architectures, so I recommend that we just block the install on x86 with an error message.

I also noticed that the x-plane SDK sample code on the website expects the headers to be installed at the root of the include prefix. This PR places the headers into `x-plane/`, which will cause the example projects to not work without amending the path. I suggest we install all the headers to the root in order to make the samples "just work" as much as possible.

I've made the above changes and pushed them to a branch: [dev/codym/5408](https://github.com/Microsoft/vcpkg/commits/dev/codym/5408). Feel free to take a look. If you like them, you can reply back here and I'll merge the changes into the PR. 😄 

I'm also still investigating what to do about the macOS frameworks. I'm not sure if they're in the correct place or not, so I'm working to find out!

xdarkradx commented:
Hi @Codiferous, thank you so much for the fixes. I changed my default triplet so I totally forgot about the x86. The changes look good to me, please feel free to merge it.

Thank you so much for your time and help.

Codiferous commented:
@xdarkradx I haven't found the answer for the framework question, but I don't think it's worth blocking this PR. I tested it out on Windows and was successfully able to build a plugin DLL (though I didn't try using it, as I don't have x-plane and I also don't want to make my coworkers jealous by hanging out with a flight simulator at work). I think even if the frameworks are placed in the wrong place on macOS, that can be addressed in another PR.

Nice work on this PR! This looks like a super cool library to have. 😄 Also, congrats on your first PR. We'd love to have you submit more!
